browser/fxr/content/fxrui.js

```diff
@@ -431,3 +438,34 @@ function finishPrompt() {
     currentPermissionRequest = null;
   }
 }
+
+// Called when navigating to another page (aState == null) or when an event
+// fires that the UI should be updated
+function permissionUpdate(aState) {  
+  // For now, only supporting microphone
+  let icon = document.getElementById("ePermsMicrophone");
+
+  // First, remove any classes that make the icon visible
+  icon.classList.remove("urlbar_device_microphone_allowed");
+  icon.classList.remove("urlbar_device_microphone_blocked");
+
+  // Next, determine if the icon should be visible and how
+  if (aState && aState.hasOwnProperty("sharing")
+     && aState.sharing === "microphone") {
+    // The microphone is in use, so show the icon in the navbar
+    icon.classList.toggle("urlbar_device_microphone_allowed", true)
+  } else {
+    let perms = SitePermissions.getAllPermissionDetailsForBrowser(browser);
```

Comment:
Would it be simpler here to just call `SitePermissions.getForPrincipal(browser.contentPrincipal, "microphone", browser)`?
Since you're only interested in the microphone permission for now.
Also, with the loop you're iterating over all permissions, even if you already found the desired one.

Reply:
ahh-- didn't know that function was available. that's definitely what i want